### PR TITLE
meal sharing homework node js

### PR DESCRIPTION
**Pull request by Bhuvaneshwari29**
Meal sharing homework completed

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,15 +1,97 @@
...
+    
+    if('maxPrice' in request.query){
+      const maxPrice = request.query.maxPrice;
+      const highPriceMeals = await knex ('meal').where ('price', '<=', maxPrice);
+      response.send(highPriceMeals);
+      return;
+    }
+    if('availableReservations' in request.query){
+      const availableReservations = (request.query.availableReservations == 'true');
+      const numberOfReservations = await knex.from('meal').join('reservations','meal.id', '=', 'reservations.id')
```
Comment (florinmaticu) [inline src/backend/api/meals.js:18]: not sure it's correct. i think the database should be `meals` and not `meal` if you generated the database from the `database_dump_meals.sql`
```suggestion
      const numberOfReservations = await knex.from('meal').join('reservations','id', '=', 'reservations.id')
```

File: src/backend/api/meals.js [hunk 2]
```diff
@@ -1,15 +1,97 @@
...
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
+
+    
+    if('maxPrice' in request.query){
+      const maxPrice = request.query.maxPrice;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
      const maxPrice = Number( request.query.maxPrice);
```

Maxprice must be a number

File: src/backend/api/meals.js [hunk 3]
```diff
@@ -1,15 +1,97 @@
...
+      const highPriceMeals = await knex ('meal').where ('price', '<=', maxPrice);
```
Comment (flom84) [inline src/backend/api/meals.js:12]:
```suggestion
      const highPriceMeals = await knex ('meals').where ('price', '<=', maxPrice);
```
 I believe the database is called `meals` and not `meal`

**Review by flom84 — CHANGES_REQUESTED** [review]

File: src/backend/api/meals.js [hunk 4]
```diff
@@ -1,15 +1,97 @@
...
+      return;
+    }
+    if('title' in request.query){
+      const title = request.query.title;
+      const mealsTitle = await knex('meal').where('title', 'like', '%' + title + '%');
+      response.send(mealsTitle);
+      return;
+    }
+    if('createdAfter' in request.query){
+      const createdAfter = request.query.createdAfter;
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
      const createdAfter = Date.parse(request.query.createdAfter);
```
be carefull with querry params . you need to check if the data is correct

File: src/backend/api/meals.js [hunk 5]
```diff
@@ -1,15 +1,97 @@
...
   } catch (error) {
     throw error;
   }
+});*/
+
+router.get('/:id', async (req,res) =>{
+  try {
+    const mealId = req.params.id;
+    const requestedMeal = await knex('meal').where ('id',mealId);
+    res.send(requestedMeal[0]);
```
Comment (flom84) [inline src/backend/api/meals.js]:
```suggestion
    res.send(requestedMeal);
```
the `id` should be unique...therefore you can get only one value or error

**Review by flom84 — CHANGES_REQUESTED** [review]

File: src/backend/api/meals.js (hunk 3 above)
Comment (Bhuvaneshwari29) [reply src/backend/api/meals.js:12]: Hi Florin,
My Table name is meal and not meals. Can i stick to the same name without changing?

File: src/backend/api/meals.js (hunk 2 above)
Comment (Bhuvaneshwari29) [reply src/backend/api/meals.js]: Yes..Thank you :-)

File: src/backend/api/meals.js (hunk 4 above)
Comment (Bhuvaneshwari29) [reply src/backend/api/meals.js]: Okay ..Thank you !!

**Comment by Bhuvaneshwari29** [comment]
Feedback implemented

**Pull request merged**